component/authenticator/src/webapp/voiceOtp.jsp

```
@@ -0,0 +1,259 @@
+            } else {
+                String authenticators = multiOptionURI.substring(authenticatorIndex + 15);
+                int authLastIndex = authenticators.indexOf("&") != -1 ? authenticators.indexOf("&") : authenticators.length();
+                authenticators = authenticators.substring(0, authLastIndex);
+                List<String> authList = new ArrayList<>(Arrays.asList(authenticators.split("%3B")));
+                if (authList.size() < 2) {
+                    isMultiAuthAvailable = false;
+                }
+                else if (authList.size() == 2 && authList.contains("backup-code-authenticator%3ALOCAL")) {
+                    isMultiAuthAvailable = false;
+                }
+            }
+        }
+        return isMultiAuthAvailable;
+    }
+%>
+
+<%
+    request.getSession().invalidate();
+    String queryString = request.getQueryString();
+    Map<String, String> idpAuthenticatorMapping = null;
+    if (request.getAttribute(Constants.IDP_AUTHENTICATOR_MAP) != null) {
+        idpAuthenticatorMapping = (Map<String, String>) request.getAttribute(Constants.IDP_AUTHENTICATOR_MAP);
+    }
+
+    String errorMessage = IdentityManagementEndpointUtil.i18n(resourceBundle,"error.retry");
+    String authenticationFailed = "false";
+
+    if (Boolean.parseBoolean(request.getParameter(Constants.AUTH_FAILURE))) {
+        authenticationFailed = "true";
+
+        if (request.getParameter(Constants.AUTH_FAILURE_MSG) != null) {
+            errorMessage = request.getParameter(Constants.AUTH_FAILURE_MSG);
+
+            if ("authentication.fail.message".equalsIgnoreCase(errorMessage)) {
+                errorMessage = IdentityManagementEndpointUtil.i18n(resourceBundle,"error.retry");
+            }
+            if (errorMessage.equalsIgnoreCase(VoiceOTPConstants.TOKEN_EXPIRED_VALUE)) {
+                errorMessage = IdentityManagementEndpointUtil.i18n(resourceBundle,"error.code.expired.resend");
+            }
```

It's better to add a fallback error message to prevent rendering non-localized content.

---

Not adding the else part here since it will hide actual error sending from the backend